In [5]:
import pandas as pd
pd.options.display.max_rows=4
pd.options.display.max_columns=100

# Get insectcodes for labels to be printed

In [6]:
df_labels_needed = pd.read_csv('NEEDS-LABELS-all-25-10-18.csv')
df_labels_needed

insectcode indb
0    INSECTGUA_0053    y
1    INSECTGUA_0310    y
..              ...  ...
577  INSECTSAI_2908    n
578  INSECTSAI_3323    n

[579 rows x 2 columns]

### Check for duplates in insectcode column

In [7]:
# Check for duplates in insectcode column
dups = df_labels_needed.insectcode.duplicated()
df_labels_needed[dups]

insectcode indb
285  INSECTGUA_2705    y

There is only one dup; let's ignore it

# Get Benita's database

In [8]:
df_db = pd.read_csv('seedinsectpredationAM.csv')

# Keep only records where insectcode is not null
df_db = df_db[df_db['insectcode'].notnull()]
df_db

Unnamed: 0     Seed.code     Tree.code  Island Tree.species.code  \
1              2  SEEDGUA_0003  TREEGUA_0003    GUAM            TRIPTR   
2              3  SEEDGUA_0003  TREEGUA_0003    GUAM            TRIPTR   
...          ...           ...           ...     ...               ...   
5694        5695  SEEDSAI_0223  TREESAI_0128  SAIPAN            PSYCMA   
5696        5697  SEEDSAI_0225  TREESAI_0129  SAIPAN            PSYCMA   

     Order.x   Family.x     Genus.x Species.x   Latitude   Longitude   Site  \
1        NaN   Rutaceae   Triphasia  trifolia  13.543554  144.933967   Anao   
2        NaN   Rutaceae   Triphasia  trifolia  13.543554  144.933967   Anao   
...      ...        ...         ...       ...        ...         ...    ...   
5694     NaN  Rubiaceae  Psychotria   mariana  15.267140  145.809016  FORBI   
5696     NaN  Rubiaceae  Psychotria   mariana  15.267120  145.809063  FORBI   

     Date..DD.MM.YYYY. Type.of.sample..F.fruit.S.seed  \
1           06/09/2017                              F   
2           06/09/2017                              F   
...                ...                            ...   
5694        27/09/2017                              F   
5696        27/09/2017                              F   

      Successional.Stage..1.immature.2.mature.3.rotten.4.all.preceding  \
1                                                     2                  
2                                                     2                  
...                                                 ...                  
5694                                                  1                  
5696                                                  1                  

     Fruits.collected.from..G.ground.T.tree Fruit.syndrome  \
1                                         T           A1.1   
2                                         T           A1.1   
...                                     ...            ...   
5694                                      T             B1   
5696                                      T             B1   

      Total.number.of.fruits.collected  Total.number.of.seeds  \
1                                    4                    5.2   
2                                    4                    5.2   
...                                ...                    ...   
5694                                48                   96.0   
5696                               108                  216.0   

      Weight.of.all.seeds.g Date.opening.seed..dd.mm.yy.  \
1                    3.7346                   18/12/1017   
2                    3.7346                   18/12/1017   
...                     ...                          ...   
5694                 5.6700                     12/19/17   
5696                14.0700                     12/19/17   

      Number.of.fruits.predated Holes.visable Frass.visable  \
1                           0.0             N             N   
2                           0.0             N             N   
...                         ...           ...           ...   
5694                       14.0             N             N   
5696                        2.0             Y             N   

      Percentage.fruits.predated      insectcode  \
1                       0.000000  INSECTGUA_1958   
2                       0.000000  INSECTGUA_1959   
...                          ...             ...   
5694                    0.291667  INSECTSAI_2807   
5696                    0.018519  INSECTSAI_1578   

     Date.emerged..for.R.only..DD.MM.YY. Species.code      Order.y Family.y  \
1                             22/11/2017       PSOSP2   Psocoptera  Unknown   
2                             22/11/2017       PSOSP2   Psocoptera  Unknown   
...                                  ...          ...          ...      ...   
5694                          09/11/2017       LEPSP4  Lepidoptera  Unknown   
5696                          30/10/2017       HYMSP7  Hymenoptera  Unknown   

     S

### Check for duplicates in insectcode column

In [9]:
# Check for duplicates in insectcode column
dups = df_db.insectcode.duplicated()
df_db[dups]

Empty DataFrame
Columns: [Unnamed: 0, Seed.code, Tree.code, Island, Tree.species.code, Order.x, Family.x, Genus.x, Species.x, Latitude, Longitude, Site, Date..DD.MM.YYYY., Type.of.sample..F.fruit.S.seed, Successional.Stage..1.immature.2.mature.3.rotten.4.all.preceding, Fruits.collected.from..G.ground.T.tree, Fruit.syndrome, Total.number.of.fruits.collected, Total.number.of.seeds, Weight.of.all.seeds.g, Date.opening.seed..dd.mm.yy., Number.of.fruits.predated, Holes.visable, Frass.visable, Percentage.fruits.predated, insectcode, Date.emerged..for.R.only..DD.MM.YY., Species.code, Order.y, Family.y, Subfamily, Genus.y, Species.y, Guild, Life.stage..L.larva.A.adult.P.pupae, Host.from.lit, Reference, Found.in.box.with, Suggested.host]
Index: []

Good; there are no dups.

# Join Tables

In [10]:
df_labels_needed.merge(right=df_db, on='insectcode')

insectcode indb  Unnamed: 0     Seed.code  \
0    INSECTGUA_0053    y         602  SEEDGUA_0041   
1    INSECTGUA_0310    y         416  SEEDGUA_0035   
..              ...  ...         ...           ...   
476  INSECTSAI_2870    y        3295  SEEDSAI_0081   
477  INSECTSAI_2884    y        4295  SEEDSAI_0164   

                                   Tree.code  Island Tree.species.code  \
0                               TREEGUA_0016    GUAM            NEISOP   
1    TREEGUA_0018, TREEGUA_0019, TREEGUA0020    GUAM            NEISOP   
..                                       ...     ...               ...   
476                             TREESAI_0044  SAIPAN            CYNORA   
477                             TREESAI_0092  SAIPAN            MELAMU   

          Order.x       Family.x      Genus.x        Species.x   Latitude  \
0     Gentianales    Apocynaceae     Ochrosia    oppositifolia  13.544063   
1     Gentianales    Apocynaceae     Ochrosia    oppositifolia  13.543936   
..            ...            ...          ...              ...        ...   
476       Fabales       Fabaceae    Cynometra        ramiflora  15.251004   
477  Malpighiales  Euphorbiaceae  Melanolepis  multiglandulosa  15.279719   

      Longitude               Site Date..DD.MM.YYYY.  \
0    144.930504               Anao        08/09/2017   
1    144.930632               Anao        08/09/2017   
..          ...                ...               ...   
476  145.798622               LTR1        19/09/2017   
477  145.807728  Last Command Post        25/09/2017   

    Type.of.sample..F.fruit.S.seed  \
0                                F   
1                                F   
..                             ...   
476                              F   
477                              F   

     Successional.Stage..1.immature.2.mature.3.rotten.4.all.preceding  \
0                                                    2                  
1                                                    2                  
..                                                 ...                  
476                                                  2                  
477                                                  2                  

    Fruits.collected.from..G.ground.T.tree Fruit.syndrome  \
0                                        G           A2.1   
1                                        G           A2.1   
..                                     ...            ...   
476                                      G             B2   
477                                      T             B2   

     Total.number.of.fruits.collected  Total.number.of.seeds  \
0                                   4                    4.0   
1                                   1                    1.0   
..                                ...                    ...   
476                                47                   94.0   
477                                66                  165.0   

     Weight.of.all.seeds.g Date.opening.seed..dd.mm.yy.  \
0                  85.1000                   18/12/1017   
1                  34.8897                   18/12/1017   
..                     ...                          ...   
476               441.6000                   12/12/1017   
477                25.3000                   18/12/1017   

     Number.of.fruits.predated Holes.visable Frass.visable  \
0                          0.0             N             N   
1                          1.0             Y             N   
..                         ...           ...           ...   
476                        6.0             Y             Y   
477                        5.0             Y             N   

     Percentage.fruits.predated Date.emerged..for.R.only..DD.MM.YY.  \
0                      0.000000                          18/09/2017   
1                      1.000000                          30/10/2017   
..                          ...                                 ...   
476    

Something wrong here. There should be 579 rows if a match in the db is found for every insectcode in the labels_needed table.

Let's find the missing insectcodes.

In [36]:
insectcodes_missing_in_db = set(df_labels_needed.insectcode.values)-set(df_db.insectcode.values)
print('There are {} insectcodes in the labels_needed table which are not in the db table:'
      .format(len(insectcodes_missing_in_db)))
insectcodes_missing_in_db

There are 101 insectcodes in the labels_needed table which are not in the db table:


{'INSECTGUA_1732',
 'INSECTGUA_1737',
 'INSECTGUA_1738',
 'INSECTGUA_1740',
 'INSECTGUA_1741',
 'INSECTGUA_1742',
 'INSECTGUA_1753',
 'INSECTGUA_1754',
 'INSECTGUA_1756',
 'INSECTGUA_1798',
 'INSECTGUA_1799',
 'INSECTGUA_1808',
 'INSECTGUA_1815',
 'INSECTGUA_1816',
 'INSECTGUA_1817',
 'INSECTGUA_1818',
 'INSECTGUA_1819',
 'INSECTGUA_1820',
 'INSECTGUA_1825',
 'INSECTGUA_1826',
 'INSECTGUA_1827',
 'INSECTGUA_1829',
 'INSECTGUA_2228',
 'INSECTGUA_2334',
 'INSECTGUA_2335',
 'INSECTGUA_2336',
 'INSECTGUA_2337',
 'INSECTGUA_2338',
 'INSECTGUA_2339',
 'INSECTGUA_2340',
 'INSECTGUA_2341',
 'INSECTGUA_2342',
 'INSECTGUA_2343',
 'INSECTGUA_2344',
 'INSECTGUA_2345',
 'INSECTGUA_2346',
 'INSECTGUA_2347',
 'INSECTGUA_2348',
 'INSECTGUA_2349',
 'INSECTGUA_2350',
 'INSECTGUA_2351',
 'INSECTGUA_2352',
 'INSECTGUA_2353',
 'INSECTGUA_2354',
 'INSECTGUA_2355',
 'INSECTGUA_2356',
 'INSECTGUA_2357',
 'INSECTGUA_2358',
 'INSECTGUA_2359',
 'INSECTGUA_2360',
 'INSECTGUA_2361',
 'INSECTGUA_2362',
 'INSECTGUA_